internal/daemon/pipeline/pipeline.go

```diff
@@ -0,0 +1,150 @@
+package pipeline
+
+import (
+	"archive/zip"
+	"bytes"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/bmatcuk/doublestar/v2"
+	"gopkg.in/yaml.v2"
+
+	"github.com/magnusbaeck/logstash-filter-verifier/v2/internal/daemon/logstashconfig"
+)
+
+type Archive struct {
+	Pipelines Pipelines
+	File      string
+	BasePath  string
+}
+
+type Pipelines []Pipeline
+
+type Pipeline struct {
+	ID      string `yaml:"pipeline.id"`
+	Config  string `yaml:"path.config"`
+	Ordered string `yaml:"pipeline.ordered"`
+	Workers int    `yaml:"pipeline.workers"`
```

Comment:
I was not able to find a reference, that these "flat keys" are a YAML feature. But in the Logstash documentation, this is mentioned (https://www.elastic.co/guide/en/logstash/current/logstash-settings-file.html), that both, nested and flat keys are supported.

From a Go code point of view, this allows me to keep the struct flat and omit the otherwise necessary nesting.

Reply:
Yeah, internally we should use the flat keys for convenience but we should support users writing files with nested keys. Since Beats supports both forms, maybe we can grab code from there? Or some other project. We wouldn't be the first ones wanting to flatten Go maps.

Reply:
This is interesting. I am not sure, how to do this, because as mentioned before, this is not a YAML feature but a special handling of Logstash. So what we might need to do is mirror both possibilities in the pipeline struct and then post-process the struct, such that the value is copied over. This raises the question, what if a user uses both ways (and does not set the same value), like this:

```yaml
- pipeline.ordered: true
  pipeline:
    ordered: false
```

Which value should have precedence in this case?